lib/puppet/provider/package/dnfmodule.rb

```diff
@@ -48,7 +49,7 @@ def self.instances
 
       packages << new(
         name: line.split[0],
-        ensure: line.split[1],
+        ensure: line =~ /\[[x]\]/ ? :disabled: line.split[1],
```

Comment:
For better readability I would move this conditional into a separate variable above, or do it like the [`flavor`](https://github.com/puppetlabs/puppet/pull/8137/files#diff-09540e265998c43bf2eb3dfd6f5628d7R44) variable for consistency.

```ruby
state = if line.include?('[x]')
          :disabled
        else
          line.split[1]
        end
```